javascript/javascript-import-and-require.md

```diff
@@ -0,0 +1,149 @@
+
+Here in Javascript, Import and Require is one of the important part and  also crucial part of it. It major useful function, It help us to divide the files into different modules and divide our code modules in systematic ways.
+
+```
+Porject:
+        assert
+          |
+          --> index.js
+        img
+          |
+          --> file.png
+        css
+          |
+          --> index.css
+          
+        index.html
+```
+Now you will get one question in your mind that How we use index.js in index.html, Now Import and Require will come is use here and using these function, you can organize your work in systematic way.
+
+## Objectives
+
+- Create understanding of import and require in user
+
+## Supplemental Materials
+
+- [JavaScript reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference)
+- [JavaScript Modules: ES6 Import and Export](https://www.youtube.com/watch?v=_3oSWwapPKQ)
+- [JavaScript Import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import)
+- [JavaScript Require](https://nodejs.org/en/knowledge/getting-started/what-is-require/)
+- [JavaScript webpack resolver](https://webpack.js.org/concepts/module-resolution/)
+
+## Lesson
+
+**Why REQUIRE and IMPORT are important in Javascript?**
+
+Code is easier to manage when it is in small, bite-size chunks. This is the thinking behind keeping functions to only one task or having files contain only a few, or one, component at a time. Here both import and require play major role in minimize and allow us to handle large number of module effectively and efficently. It is mainly used to get modules from different dictionary and use them. If you have a complex app and have to scroll through hundreds or thousands of lines of code, then the job of debugging or just understanding the app becomes that much harder.
+
+Javascript helps us out with this by having ‘imports’ and ‘require’. This is how you can write code in one file and share that code so it can be used by another file or files
+
+**What are the differences between IMPORT and REQUIRE?**
+|Import|Require|
+|---|---|
+|As import remains in stage three and not<br> enforced by browsers natively, we’re unable to<br> run any performance take a look at.|Presently once you use import in your code,<br> your transpilers it back to require,<br> the commonJS modeling system. <br>Therefore for nowadays each is the same|
+|Though there aren’t any performance profit at the instant,<br> however, I’ll still counsel to use import over<br> require because it’s about to be native in JS|As a result of import is native therefore<br>require doesn’t perform higher as compare to import|

```

Comment:
Is this text from here?: https://www.educba.com/require-vs-import/

Reply:
can you make it more clear that you are quoting text to create the lesson? Always cite your sources to avoid plagiarism.

If you'd like people to read a lot of an article,  just tell people to read it and give them the link, and any other notes.   If it is a short quote, just be sure to add quotes, or give it an article formatting, and provide a link at the end like this [from Google](www.google.com)